### ISSUE TITLE
line and underscore blinking visual bug

### ISSUE THREAD

**Issue by pcroland** [mintty/wsltty#255]
If I set the cursor type to line or underscore the first blink creates a permanent pixel at the bottom that won't move with most of the cursor:
![image1](https://widevine.is-a.fail/T3yp8L.png) ![image2](https://widevine.is-a.fail/KKEXwZ.png)
font: MesloLGS NF, 9pt

**Comment by mintty** [comment]
Cannot reproduce with Meslo NG N. There are a bunch of Meslo fonts around, also additional versions patched for "Powerline".
Please provide a precise download link for a version you use. Does it happen with other fonts? Any specific config settings, specifically RowSpacing?

**Comment by pcroland** [comment]
I grabbed the fonts from here: [https://github.com/romkatv/powerlevel10k](url)
Config:
```# To use common configuration in %APPDATA%\mintty, simply remove this file
ThemeFile=rosipov
Font=MesloLGS NF
CursorType=line
OpaqueWhenFocused=yes
Transparency=off
Term=xterm
BellFile=ambient-drop
FontHeight=10
FontSmoothing=default
Columns=160
Rows=50
Scrollbar=right
Language=en_US
EmojiPlacement=middle
Emojis=apple
```

**Comment by mintty** [comment]
Does it happen only with font size 9pt as you said first or also with the configured 10pt?
Does it happen if you start with the quoted config (CursorType=line) or only if you switch from block to line cursor?
As you didn't provide a download link, I assume you refer to the files in section "Manual font installation". Not reproduced.

**Comment by pcroland** [comment]
It happens with any font size and it always happens, just opening wsltty also triggers it. It also appears when I do `clear`:
![image](https://x0.at/Tc8.mp4)

**Comment by mintty** [comment]
I assume it also does not depend on the terminal size. Try e.g. 24x80. So what's your screen resolution (DPI)?

**Comment by mintty** [comment]
Also it might be related to your specific prompt setting, maybe some special characters in it. Please make a screen log (mintty -l ...).

**Comment by pcroland** [comment]
Yes, it does not depend on terminal size. Screen res is basic 1920x1080 with default (100%) scaling.

**Comment by mintty** [comment]
And try with simple prompt (PS1=.) as a cross-test.

**Comment by mintty** [comment]
Also, what's the "Leading" value reported in Options - Fonts?

**Comment by pcroland** [comment]
![i](https://widevine.is-a.fail/OFMu0j.png)

**Comment by mintty** [comment]
I think we checked everything except the possibility of an artefact caused by some weird character handling for something in your prompt. So asking again: what happens after PS1=. and if that changes things, please put your prompt into a text file so the effect becomes apparent after `cat file` and upload that here please.

**Comment by mintty** [comment]
I should have asked one more basic thing: what's your version?

**Comment by pcroland** [comment]
`PS1=.` does not do anything, the pixel appears the same way.
Log file: [https://privatebin.at/?08e3c950502b0342#3KMzsyt826czyBtGmqyWykk5TLAPe5C2WLZDfrPDxaFr](https://privatebin.at/?08e3c950502b0342#3KMzsyt826czyBtGmqyWykk5TLAPe5C2WLZDfrPDxaFr)
zsh theme: powerlevel10k
I'm using the latest build from /releases (3.3.0)

**Comment by pcroland** [comment]
It doesn't happen with bash, just with zsh (with powelevel10k theme). Is it possible that the first blink is bold and somehow it renders a glitched pixel at the next line? At least that's how it seems.

**Comment by mintty** [comment]
The issue is triggered by the character U+F017 alone.

**Comment by mintty** [comment]
There seems to be some utterly weird interference between specific characters (of specific fonts?) and the Windows Rectangle function which is used to paint the cursor. The line cursor would be one column wider instantly and become narrower immediately (made visible by some delay) and leave a pixel on that occasion. I tweaked cursor painting by using the FillRect function instead which does not seem to exhibit such broken phenomenon.
Applied patch for line cursor. I could not reproduce the issue with an underscore cursor, though.

**Comment by pcroland** [comment]
Thanks a lot! I'll update asap if a new version comes out.

**Comment by mintty** [comment]
Released 3.4.1.
